File: OpenRobertaServer/src/test/resources/crossCompilerTests/robotSpecific/_expected/targetLanguage/ev3dev/___listInitializations.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
        'B':Hal.makeLargeMotor(ev3dev.OUTPUT_B, 'on', 'foreward'),
    },
    'sensors': {
        '1':Hal.makeTouchSensor(ev3dev.INPUT_1),
        '2':Hal.makeGyroSensor(ev3dev.INPUT_2),
        '3':Hal.makeColorSensor(ev3dev.INPUT_3),
        '4':Hal.makeUltrasonicSensor(ev3dev.INPUT_4),
    },
}
hal = Hal(_brickConfiguration)

___numbers = [0, hal.getUltraSonicSensorDistance('4'), hal.getMotorTachoValue('B', 'degree'), hal.getMotorTachoValue('B', 'rotation'), hal.getMotorTachoValue('B', 'distance'), hal.getGyroSensorValue('2', 'angle'), hal.getGyroSensorValue('2', 'rate'), hal.getTimerValue(1)]
___strings = [True, False, hal.isPressed('1'), hal.isKeyPressed('enter'), hal.getUltraSonicSensorPresence('4'), True == hal.isKeyPressed('enter'), True != hal.isKeyPressed('enter'), hal.getUltraSonicSensorDistance('4') < 0, hal.getUltraSonicSensorDistance('4') <= 0, hal.getUltraSonicSensorDistance('4') > 0, hal.getUltraSonicSensorDistance('4') >= 0, True and hal.isKeyPressed('enter'), True or hal.isKeyPressed('enter')]
___booleans = ["", "".join(str(arg) for arg in ["", hal.getColorSensorColour('3'), [""], True, 0, 'white'])]
___colors = ['white', hal.getColorSensorColour('3')]
___connections = [hal.waitForConnection(), hal.establishConnectionTo("")]
___item = [5] * 5
___item2 = [True] * 5
___item3 = [""] * 5
___item4 = ['blue'] * 5
___item5 = [hal.waitForConnection()] * 5
___item6 = [5 + float(1.2)] * 5
___item7 = [True] * 5
___item8 = ["".join(str(arg) for arg in ["", hal.getColorSensorColour('3')])] * 5
___item9 = [hal.getColorSensorColour('3')] * 5
def run():
    global ___numbers, ___strings, ___booleans, ___colors, ___connections, ___item, ___item2, ___item3, ___item4, ___item5, ___item6, ___item7, ___item8, ___item9

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()

File: OpenRobertaServer/src/test/resources/crossCompilerTests/robotSpecific/_expected/targetLanguage/ev3lejosv1/___listInitializations.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class null {
    private static Configuration brickConfiguration;
    
    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>(Arrays.asList(new UsedSensor(SensorPort.S4, SensorType.ULTRASONIC, UltrasonicSensorMode.DISTANCE), new UsedSensor(SensorPort.S2, SensorType.GYRO, GyroSensorMode.ANGLE), new UsedSensor(SensorPort.S2, SensorType.GYRO, GyroSensorMode.RATE), new UsedSensor(SensorPort.S1, SensorType.TOUCH, TouchSensorMode.TOUCH), new UsedSensor(SensorPort.S4, SensorType.ULTRASONIC, UltrasonicSensorMode.PRESENCE), new UsedSensor(SensorPort.S3, SensorType.COLOR, ColorSensorMode.COLOUR)));
    private Hal hal = new Hal(brickConfiguration, usedSensors);
    
    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .addActor(ActorPort.B, new Actor(ActorType.LARGE, true, DriveDirection.FOREWARD, MotorSide.RIGHT))
                .addSensor(SensorPort.S1, new Sensor(SensorType.TOUCH))
                .addSensor(SensorPort.S2, new Sensor(SensorType.GYRO))
                .addSensor(SensorPort.S3, new Sensor(SensorType.COLOR))
                .addSensor(SensorPort.S4, new Sensor(SensorType.ULTRASONIC))
                .build();
            
            new null().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }
    
    ArrayList<Float> ___numbers = new ArrayList<>(Arrays.asList((float) 0, (float) hal.getUltraSonicSensorDistance(SensorPort.S4), (float) hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.DEGREE), (float) hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.ROTATION), (float) hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.DISTANCE), (float) hal.getGyroSensorAngle(SensorPort.S2), (float) hal.getGyroSensorRate(SensorPort.S2), (float) hal.getTimerValue(1)));
    ArrayList<Boolean> ___strings = new ArrayList<>(Arrays.<Boolean>asList(true, false, hal.isPressed(SensorPort.S1), hal.isPressed(BrickKey.ENTER), hal.getUltraSonicSensorPresence(SensorPort.S4), true == hal.isPressed(BrickKey.ENTER), true != hal.isPressed(BrickKey.ENTER), hal.getUltraSonicSensorDistance(SensorPort.S4) < 0, hal.getUltraSonicSensorDistance(SensorPort.S4) <= 0, hal.getUltraSonicSensorDistance(SensorPort.S4) > 0, hal.getUltraSonicSensorDistance(SensorPort.S4) >= 0, true && hal.isPressed(BrickKey.ENTER), true || hal.isPressed(BrickKey.ENTER)));
    ArrayList<String> ___booleans = new ArrayList<>(Arrays.<String>asList("", String.valueOf("") + String.valueOf(hal.getColorSensorColour(SensorPort.S3)) + String.valueOf(new ArrayList<>(Arrays.<String>asList(""))) + String.valueOf(true) + String.valueOf(0) + String.valueOf(PickColor.WHITE)));
    ArrayList<PickColor> ___colors = new ArrayList<>(Arrays.<PickColor>asList(PickColor.WHITE, hal.getColorSensorColour(SensorPort.S3)));
    ArrayList<NXTConnection> ___connections = new ArrayList<>(Arrays.<NXTConnection>asList(hal.waitForConnection(), hal.establishConnectionTo("")));
    ArrayList<Float> ___item = new ArrayList<>(Collections.nCopies( (int) 5,  (float) 5));
    ArrayList<Boolean> ___item2 = new ArrayList<>(Collections.nCopies( (int) 5, true));
    ArrayList<String> ___item3 = new ArrayList<>(Collections.nCopies( (int) 5, ""));
    ArrayList<PickColor> ___item4 = new ArrayList<>(Collections.nCopies( (int) 5, PickColor.BLUE));
    ArrayList<NXTConnection> ___item5 = new ArrayList<>(Collections.nCopies( (int) 5, hal.waitForConnection()));
    ArrayList<Float> ___item6 = new ArrayList<>(Collections.nCopies( (int) 5, 5 + ((float) 1.2)));
    ArrayList<Boolean> ___item7 = new ArrayList<>(Collections.nCopies( (int) 5, true));
    ArrayList<String> ___item8 = new ArrayList<>(Collections.nCopies( (int) 5, String.valueOf("") + String.valueOf(hal.getColorSensorColour(SensorPort.S3))));
    ArrayList<PickColor> ___item9 = new ArrayList<>(Collections.nCopies( (int) 5, hal.getColorSensorColour(SensorPort.S3)));
    
    public void run() throws Exception {
    }

}